run-tests.sh

```diff
@@ -1,29 +1,83 @@
 #!/bin/bash
 #
 # This file is part of REANA.
-# Copyright (C) 2017, 2018 CERN.
+# Copyright (C) 2017, 2018, 2019, 2020 CERN.
 #
 # REANA is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
+# Quit on errors
+set -o errexit
+
+# Quit on unbound symbols
+set -o nounset
+
+# Verify that db container is running before continuing
+_check_ready() {
+    RETRIES=40
+    while ! $2
+    do
+        echo "==> [INFO] Waiting for $1, $((RETRIES--)) remaining attempts..."
+        sleep 2
+        if [ $RETRIES -eq 0 ]
+        then
+            echo "==> [ERROR] Couldn't reach $1"
+            exit 1
+        fi
+    done
+}
+
+_db_check() {
+    docker exec --user postgres postgres__reana-job-controller bash -c "pg_isready" &>/dev/null;
+}
+
+clean_old_db_container() {
+    OLD="$(docker ps --all --quiet --filter=name=postgres__reana-job-controller)"
+    if [ -n "$OLD" ]; then
+        echo '==> [INFO] Cleaning old DB container...'
+        docker stop postgres__reana-job-controller
+    fi
+}
+
+start_db_container() {
+    echo '==> [INFO] Starting DB container...'
+    docker run --rm --name postgres__reana-job-controller -p 5432:5432 -e POSTGRES_PASSWORD=mysecretpassword -d postgres:9.6.2
+    _check_ready "Postgres" _db_check
+}
+
+stop_db_container() {
+    echo '==> [INFO] Stopping DB container...'
+    docker stop postgres__reana-job-controller
+}
+
+check_black() {
+    echo "echo '==> [INFO] Checking Black compliance...'"
+    echo "black --check ."
+}
+
 COMPONENT_NAME=reana-job-controller
 DOCKER_IMAGE_NAME=reanahub/$COMPONENT_NAME
 PLATFORM="$(python -c 'import platform; print(platform.system())')"
 
+clean_old_db_container
+start_db_container
+db_container_ip=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' postgres__reana-job-controller)
+SQLALCHEMY_URI=postgresql+psycopg2://postgres:mysecretpassword@$db_container_ip/postgres
+
 RUN_TESTS="pydocstyle reana_job_controller &&
-black --check . &&
+$(check_black) &&
```

Comment:
`check_black` instead of `$(check_black)`?

Reply:
This is run inside the r-j-controller container so `check_black` function is unknown, that's why I'm echoing the contents of `check_black` so it can run them.